src/java.base/share/classes/jdk/internal/util/Architecture.java

```diff
@@ -154,6 +155,14 @@ public static boolean isS390() {
         return PlatformProps.TARGET_ARCH_IS_S390;
     }
 
+    /**
+     * {@return {@code true} if the current architecture is PPC, big-endian}
+     */
+    @ForceInline
+    public static boolean isPPC() {
+        return PlatformProps.TARGET_ARCH_IS_PPC;
+    }
```

Comment:
Maybe `isPPC32()` would be a better name (also `PPC32` above). Note that hotspot uses `PPC` for all PPC platforms and has the more specific macros `PPC32` and `PPC64`.

Reply:
Yeah, we can technically change to `PPC32`. But the current thing follows what build system uses (`ppc`) and what would be used as `os.arch` because of that. So, choosing which way to deviate: towards build-system/java-props or towards hotspot macros, I think what current PR does is the lesser evil.

Reply:
One might expect `isPPC()` to return true on PPC64 because it technically is. This is the only drawback I see.

Reply:
I don't have a strong opinion here, so if you want PPC32, you got it in new commit :)